armi/reactor/blocks.py

```diff
@@ -1650,6 +1650,39 @@ def getDuctOP(self):
         duct = self.getComponent(Flags.DUCT, exact=True)
         return duct.getDimension("op")
 
+    def setB10VolParam(self, heightHot):
+        """Set the b.p.initialB10ComponentVol param according to the volume of boron-10 containing components."""
+        b10Comps = []
+        potentialC = None
+        for c in self:
+            if c.getNumberDensity("B10"):
+                potentialC = c
+                b10Comps.append(potentialC)
+                tHot = potentialC.temperatureInC
+                tCold = potentialC.inputTemperatureInC
+                mat = potentialC.material
+        if potentialC is None:
+            return
+
+        # calc volume of boron components
+        coldArea = 0.0
+        for c in b10Comps:
+            if (
+                tHot != c.temperatureInC
+                or tCold != c.inputTemperatureInC
+                or mat != c.material
+            ):
+                runLog.warning(
+                    f"More than 1 component found  in {b.name} with different B10 material"
+                    f" or temperature, using {potentialC} params",
+                    single=True,
+                )
+            coldArea += c.getArea(cold=True)
+
+        coldFactor = potentialC.getThermalExpansionFactor() if heightHot else 1
```

Comment:
i still think i want the warning because even if its cold but they are differnt temps, it means we will have mass flow of boron. ill clarify in the warning